### ISSUE TITLE
OpenCV 4.5.2 conda package generation regression

### ISSUE THREAD

**Issue by traversaro** [robotology/robotology-superbuild#711]

**Comment by traversaro** [comment]
See https://github.com/robotology/robotology-superbuild/runs/2502612360?check_suite_focus=true . When generating the icub-main package on conda/macOS, the builds fails with:
~~~
2021-05-04T19:07:06.3453530Z [310/424] Linking CXX executable bin/stereoCalib
2021-05-04T19:07:06.3454040Z FAILED: bin/stereoCalib 
2021-05-04T19:07:06.3471400Z : && $BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -std=c++14 -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/icub-main-1.19.2 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.13 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,$PREFIX/lib -L$PREFIX/lib src/tools/stereoCalib/CMakeFiles/stereoCalib.dir/src/main.cpp.o src/tools/stereoCalib/CMakeFiles/stereoCalib.dir/src/stereoCalibModule.cpp.o src/tools/stereoCalib/CMakeFiles/stereoCalib.dir/src/stereoCalibThread.cpp.o -o bin/stereoCalib  $PREFIX/lib/libopencv_gapi.4.5.1.dylib  $PREFIX/lib/libopencv_stitching.4.5.1.dylib  $PREFIX/lib/libopencv_alphamat.4.5.1.dylib  $PREFIX/lib/libopencv_aruco.4.5.1.dylib  $PREFIX/lib/libopencv_bgsegm.4.5.1.dylib  $PREFIX/lib/libopencv_bioinspired.4.5.1.dylib  $PREFIX/lib/libopencv_ccalib.4.5.1.dylib  $PREFIX/lib/libopencv_dnn_objdetect.4.5.1.dylib  $PREFIX/lib/libopencv_dnn_superres.4.5.1.dylib  $PREFIX/lib/libopencv_dpm.4.5.1.dylib  $PREFIX/lib/libopencv_face.4.5.1.dylib  $PREFIX/lib/libopencv_freetype.4.5.1.dylib  $PREFIX/lib/libopencv_fuzzy.4.5.1.dylib  $PREFIX/lib/libopencv_hdf.4.5.1.dylib  $PREFIX/lib/libopencv_hfs.4.5.1.dylib  $PREFIX/lib/libopencv_img_hash.4.5.1.dylib  $PREFIX/lib/libopencv_intensity_transform.4.5.1.dylib  $PREFIX/lib/libopencv_line_descriptor.4.5.1.dylib  $PREFIX/lib/libopencv_mcc.4.5.1.dylib  $PREFIX/lib/libopencv_quality.4.5.1.dylib  $PREFIX/lib/libopencv_rapid.4.5.1.dylib  $PREFIX/lib/libopencv_reg.4.5.1.dylib  $PREFIX/lib/libopencv_rgbd.4.5.1.dylib  $PREFIX/lib/libopencv_saliency.4.5.1.dylib  $PREFIX/lib/libopencv_stereo.4.5.1.dylib  $PREFIX/lib/libopencv_structured_light.4.5.1.dylib  $PREFIX/lib/libopencv_superres.4.5.1.dylib  $PREFIX/lib/libopencv_surface_matching.4.5.1.dylib  $PREFIX/lib/libopencv_tracking.4.5.1.dylib  $PREFIX/lib/libopencv_videostab.4.5.1.dylib  $PREFIX/lib/libopencv_xfeatures2d.4.5.1.dylib  $PREFIX/lib/libopencv_xobjdetect.4.5.1.dylib  $PREFIX/lib/libopencv_xphoto.4.5.1.dylib  lib/libiKin.a  $PREFIX/lib/libopencv_phase_unwrapping.4.5.1.dylib  $PREFIX/lib/libopencv_optflow.4.5.1.dylib  $PREFIX/lib/libopencv_highgui.4.5.1.dylib  $PREFIX/lib/libopencv_datasets.4.5.1.dylib  $PREFIX/lib/libopencv_plot.4.5.1.dylib  $PREFIX/lib/libopencv_text.4.5.1.dylib  $PREFIX/lib/libopencv_videoio.4.5.1.dylib  $PREFIX/lib/libopencv_ml.4.5.1.dylib  $PREFIX/lib/libopencv_shape.4.5.1.dylib  $PREFIX/lib/libopencv_ximgproc.4.5.1.dylib  $PREFIX/lib/libopencv_video.4.5.1.dylib  $PREFIX/lib/libopencv_dnn.4.5.1.dylib  $PREFIX/lib/libopencv_imgcodecs.4.5.1.dylib  $PREFIX/lib/libopencv_objdetect.4.5.1.dylib  $PREFIX/lib/libopencv_calib3d.4.5.1.dylib  $PREFIX/lib/libopencv_features2d.4.5.1.dylib  $PREFIX/lib/libopencv_flann.4.5.1.dylib  $PREFIX/lib/libopencv_photo.4.5.1.dylib  $PREFIX/lib/libopencv_imgproc.4.5.1.dylib  $PREFIX/lib/libopencv_core.4.5.1.dylib  -lopencv_wechat_qrcode  lib/libctrlLib.a  $PREFIX/lib/libYARP_init.3.4.3.dylib  $PREFIX/lib/libYARP_dev.3.4.3.dylib  $PREFIX/lib/libYARP_math.3.4.3.dylib  $PREFIX/lib/libYARP_sig.3.4.3.dylib  $PREFIX/lib/libYARP_os.3.4.3.dylib  $PREFIX/lib/libgsl.dylib  $PREFIX/lib/libgslcblas.dylib  $PREFIX/lib/libipopt.dylib && :
2021-05-04T19:07:06.3486380Z ld: library not found for -lopencv_wechat_qrcode
2021-05-04T19:07:06.3487470Z clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
2021-05-04T19:07:06.3488380Z [311/424] Building CXX object src/tools/depth2kin/CMakeFiles/depth2kin.dir/PointReq.cpp.o
2021-05-04T19:07:06.3489750Z [312/424] Building CXX object src/tools/depth2kin/CMakeFiles/depth2kin.dir/src/nlp.cpp.o
2021-05-04T19:07:06.3490640Z [313/424] Building CXX object src/tools/depth2kin/CMakeFiles/depth2kin.dir/src/module.cpp.o
2021-05-04T19:07:06.3491580Z [314/424] Building CXX object src/tools/depth2kin/CMakeFiles/depth2kin.dir/depth2kin_IDL.cpp.o
2021-05-04T19:07:06.3492320Z ninja: build stopped: subcommand failed.
2021-05-04T19:07:06.3492930Z Traceback (most recent call last):
2021-05-04T19:07:06.3494860Z   File "/usr/local/miniconda/envs/test/bin/conda-build", line 11, in <module>
2021-05-04T19:07:06.3495810Z     sys.exit(main())
2021-05-04T19:07:06.3497270Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 481, in main
2021-05-04T19:07:06.3498320Z     execute(sys.argv[1:])
2021-05-04T19:07:06.3499810Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 470, in execute
2021-05-04T19:07:06.3501110Z     outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
2021-05-04T19:07:06.3502750Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
2021-05-04T19:07:06.3503780Z     return build_tree(
2021-05-04T19:07:06.3505230Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/build.py", line 3068, in build_tree
2021-05-04T19:07:06.3506370Z     packages_from_this = build(metadata, stats,
2021-05-04T19:07:06.3507860Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/build.py", line 2191, in build
2021-05-04T19:07:06.3509590Z     utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
2021-05-04T19:07:06.3511310Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/utils.py", line 412, in check_call_env
2021-05-04T19:07:06.3512950Z     return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
2021-05-04T19:07:06.3514720Z   File "/usr/local/miniconda/envs/test/lib/python3.8/site-packages/conda_build/utils.py", line 392, in _func_defaulting_env_to_os_environ
2021-05-04T19:07:06.3516150Z     raise subprocess.CalledProcessError(proc.returncode, _args)
2021-05-04T19:07:06.3519190Z subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/usr/local/miniconda/envs/test/conda-bld/icub-main_1620154618464/work/conda_build.sh']' returned non-zero exit status 1.
~~~

This is probably some kind of regression caused by https://github.com/conda-forge/opencv-feedstock/pull/271 and the specific CMake OpenCV machinery in icub-main in https://github.com/robotology/icub-main/blob/26c74355c9d6463762352f41f8f1a1df593844ca/src/tools/stereoCalib/CMakeLists.txt#L17 , as instead compilation against OpenCV works fine in YARP.

**Comment by traversaro** [comment]
We should solve this, as 4.5.1 binaries are onlly available with protobuf 3.15, and as the transition to 3.16 is ongoing (see https://github.com/conda-forge/gazebo-feedstock/pull/83) so pinning opencv 4.5.1 effectively means that we are pinning protobuf to 3.15 and we are not using the latest binaries.
